src/types/dict.rs

```diff
@@ -157,19 +157,17 @@ impl PyDict {
     /// returns `Ok(None)` if item is not present, or `Err(PyErr)` if an error occurs.
     ///
     /// To get a `KeyError` for non-existing keys, use `PyAny::get_item_with_error`.
-    #[cfg(not(PyPy))]
     pub fn get_item_with_error<K>(&self, key: K) -> PyResult<Option<&PyAny>>
     where
         K: ToPyObject,
     {
         unsafe {
             let ptr =
                 ffi::PyDict_GetItemWithError(self.as_ptr(), key.to_object(self.py()).as_ptr());
```

Comment:
Admittedly unrelated to change in question, but I was wondering whether this line becomes a potential use-after-free when we drop the pool? If the owning pointer returned by `key.to_object(self.py())` is dropped while the result of `.as_ptr()` is passed to `PyDict_GetItemWithError`? Is there some sort of lifetime extension for the temporary in play? Probably not as the raw point does not carry a lifetime?

Reply:
Good question, I believe it is sound due to the temporaries all being held until the end of the statement.

Actually this is against the style which @mejrs and I agreed in https://github.com/PyO3/pyo3/pull/2312#discussion_r860233477, so let me adjust this to bind the temporary on a new line, just to be sure.

Reply:
Debloating the generics is certainly a nice flourish.